src/core/thread/router_table.cpp

```diff
@@ -115,8 +115,9 @@ void RouterTable::UpdateAllocation(void)
     // build index map
     for (uint8_t i = 0; i <= Mle::kMaxRouterId; i++)
     {
-        if (IsAllocated(i) && mActiveRouterCount < Mle::kMaxRouters)
+        if (IsAllocated(i))
         {
+            assert(mActiveRouterCount < Mle::kMaxRouters);
```

Comment:
I think this assert will cause issues with fuzz testing. Note that the [MLE security checks are disabled](https://github.com/openthread/openthread/blob/2057a141ef75247197634d709bff4789d34c02d3/src/core/thread/mle.cpp#L2579) during fuzz testing.

Reply:
Ok. Let's keep the original code. Closing this JIRA